### PR TITLE
implemented proxy logic, added package-lock file;

### PR DESCRIPTION
**Pull request by mcarneiro**
Made a quick proxy implementation so I could work remotelly. Didn't update any tests or documentation, but I can say it's working. So please take a look.

Cheers

### CODE REVIEW

**Comment by mcarneiro** [comment]
related to issue #4

File: lib/crex.js [hunk 1]
```diff
@@ -39,7 +41,7 @@ CrEx.prototype.getUrl = function () {
 
 CrEx.prototype.getAddress = function () {
 	var url = this.url;
-	return this.port !== '' ? url + ':' + this.port : url;
+	return (this.port !== '' ? url + ':' + this.port : url) + (this.proxy ? ' (proxy: ' + this.proxy + ')' : '');
```
Comment (mcarneiro) [inline lib/crex.js]: Saw that this function is only used for console purposes. That's why I concat the proxy info.

File: lib/request.js [hunk 2]
```diff
@@ -1,4 +1,9 @@
 var request = require('superagent');
+require('superagent-proxy')(request);
+
+var checkProxy = function(req, proxy) {
+	return proxy ? req.proxy(proxy) : req;
```
Comment (mateuszluczak) [inline lib/request.js]: I've tried to google other way around this, but looks like ``superagent-proxy`` can't handle empty proxy with ``req.proxy``, so this approach seems 👍

**Comment by mcarneiro** [comment]
@mateuszluczak just added a `proxy` key inside `env` prop in `auth.json` file (basically what CrEx constuctor receives). So you could have, for example, `dev.proxy` with values that [node-proxy-agent](https://github.com/TooTallNate/node-proxy-agent) used by `superagent-proxy` accepts:

```
{
  "dev": {
    "user": "admin",
    "password": "123456",
    "url": "127.0.0.1",
    "port": "4000",
    "proxy": "http://18:222.0.18:3128"
  }
}
```

Just tested with `http` proxy. Made a quick try with `pac` file, but it didn't work. Maybe `node-proxy-agent` issue.

**Pull request merged**